caddyhttp/httpserver/replacer.go

```diff
@@ -309,6 +309,12 @@ func (r *replacer) getSubstitution(key string) string {
 	case "{dir}":
 		dir, _ := path.Split(r.request.URL.Path)
 		return dir
+	case "{user}":
+		user, _ := r.request.Context().Value(RemoteUserCtxKey).(string)
+		if user == "" {
+			user = "-"
```

Comment:
I don't think we need an empty value here; any middleware that wants a default value for empty placeholders will pass in what that value should be.